src/states/adult.rs

```diff
@@ -501,17 +528,29 @@ impl Approved for Adult {
             })
         } else {
             debug!("{} - Unhandled SectionInfo event", self);
+
+            // Need to pop the relocate queue even though we are not going to vote. Otherwise it
+            // could get out of sync with the rest of the section when we transition to elder.
+            if elders_info.prefix().matches(self.name()) {
+                let _ = self.chain.poll_relocation();
```

Comment:
Should it be within an if? what happen after a split when one side will get its elder_info, and the other side will get the same elder_info that is not theirs?

Reply:
Good point, but I think it still works as is, although possibly not optimally. What happens is that one half of the section processes the next relocation immediately but the other half only on the next self-SectionInfo. Everything is still consistent within each section though. 

I had an idea (and I think you had a similar one) to use the churn_event_backlog also for relocations which I think would solve this issue too (and also make things a bit cleaner). Something for a followup PR perhaps?

Reply:
Yes, make sense